test/fluree/db/storage/s3_indexing_test.clj

```diff
@@ -0,0 +1,134 @@
+(ns fluree.db.storage.s3-indexing-test
+  "Test S3 storage with indexing"
+  (:require [clojure.string :as str]
+            [clojure.test :refer [deftest is testing]]
+            [cognitect.aws.client.api :as aws]
+            [fluree.db.api :as fluree]))
+
+(defn setup-env []
+  (System/setProperty "aws.accessKeyId" "test")
+  (System/setProperty "aws.secretAccessKey" "test")
+  (System/setProperty "aws.region" "us-east-1"))
+
+(defn list-s3-objects [bucket prefix]
+  (let [client (aws/client {:api :s3
+                            :endpoint-override {:protocol :http
+                                                :hostname "localhost"
+                                                :port 4566}})
+        result (aws/invoke client {:op :ListObjectsV2
+                                   :request {:Bucket bucket
+                                             :Prefix prefix}})]
+    (:Contents result)))
+
+(defn get-index-files-by-type
+  "Categorize index files by type (root, post, spot, tspo, opst)"
+  [objects]
+  (let [index-files (filter #(str/includes? (:Key %) "/index/") objects)]
+    {:root (filter #(str/includes? (:Key %) "/index/root/") index-files)
+     :post (filter #(str/includes? (:Key %) "/index/post/") index-files)
+     :spot (filter #(str/includes? (:Key %) "/index/spot/") index-files)
+     :tspo (filter #(str/includes? (:Key %) "/index/tspo/") index-files)
+     :opst (filter #(str/includes? (:Key %) "/index/opst/") index-files)
+     :all index-files}))
+
+(deftest ^:integration ^:pending test-s3-indexing
+  (testing "S3 storage with indexing triggers and validation"
+    ;; Marking as pending until LocalStack setup is improved in CI
+    (setup-env)
+    (let [bucket "fluree-test"
+          prefix "indexing"
+          ledger-name (str "index-test-" (System/currentTimeMillis))
+          context {"@vocab" "https://ns.flur.ee/ledger#"
+                   "ex" "http://example.org/ns/"}]
+
+      (testing "Index creation with low thresholds"
+        (let [conn @(fluree/connect-s3
+                     {:s3-bucket bucket
+                      :s3-prefix prefix
+                      :s3-endpoint "http://localhost:4566"
+                      :defaults {:indexing {:reindex-min-bytes 100
+                                            :reindex-max-bytes 1000}}})]
+
+          (testing "Ledger creation and initial transaction"
+            (let [ledger @(fluree/create conn ledger-name)
+                  db0 (fluree/db ledger)
+                  data1 [{"@id" "ex:alice" "ex:name" "Alice" "ex:age" 30}
+                         {"@id" "ex:bob" "ex:name" "Bob" "ex:age" 25}]
+                  db1 @(fluree/stage db0 {"@context" context "insert" data1})]
+
+              (is (some? ledger) "Ledger should be created successfully")
+              (is (some? db1) "Data should stage successfully")
+
+              @(fluree/commit! ledger db1)
+
+              ;; Wait for async indexing
```

Comment:
I know all too well the difficulties of dealing with localstack so I'm not going to ask that this sleep be removed, but could you add a FIXME comment to find a more deterministic solution to waiting for the indexes to kick off and finish?